streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -481,7 +481,6 @@ void commit(final boolean startNewTransaction, final Map<TopicPartition, Long> p
             final long offset = entry.getValue() + 1;
             final long partitionTime = partitionTimes.get(partition);
             consumedOffsetsAndMetadata.put(partition, new OffsetAndMetadata(offset, encodeTimestamp(partitionTime)));
-            stateMgr.putOffsetLimit(partition, offset);
```

Comment:
This line is not needed since for stream task we only need to put offset limit once, which is before the restoration. During normal processing we do not need to set offset limit any more.

Reply:
+1. That should not be needed.